# Testing Results

I will edit this post on the fly 

## Error 1: ERROR: Got unknown keyword arguments "version"
When doing a fresh clone and building the MBS Transport Function the following error occurs:

```
subprojects/rt-libflute/meson.build:127:0: ERROR: Got unknown keyword arguments "version"
```

### Quick fix:
Disable unit tests in libflute `CMakeLists.txt`

``` 
# ---- Tests Subdirectory (optional) ----
option(BUILD_TESTING "Build unit tests" OFF)
if (BUILD_TESTING AND NOT DEFINED GTEST_DISABLE)
  add_subdirectory(tests)
endif()
```

### Clean fix_

## Error 2: FAILED: src/mbstf/open5gs-mbstfd 

```
FAILED: src/mbstf/open5gs-mbstfd 
c++  -o src/mbstf/open5gs-mbstfd src/mbstf/open5gs-mbstfd.p/App.cc.o src/mbstf/open5gs-mbstfd.p/.._.._subprojects_open5gs_src_main.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,--whole-archive -Wl,--start-group src/mbstf/libmbstf.a -Wl,--no-whole-archive '-Wl,-rpath,$ORIGIN/../../subprojects/open5gs/lib/sbi:$ORIGIN/../../subprojects/open5gs/lib/crypt:$ORIGIN/../../subprojects/open5gs/lib/proto:$ORIGIN/../../subprojects/open5gs/lib/core:$ORIGIN/../../subprojects/open5gs/lib/app:$ORIGIN/../../subprojects/open5gs/lib/sbi/openapi:/usr/local/lib/x86_64-linux-gnu' subprojects/open5gs/lib/sbi/libogssbi.so.2.6.4 subprojects/open5gs/lib/crypt/libogscrypt.so.2.6.4 subprojects/open5gs/lib/proto/libogsproto.so.2.6.4 subprojects/open5gs/lib/core/libogscore.so.2.6.4 subprojects/open5gs/lib/app/libogsapp.so.2.6.4 subprojects/open5gs/lib/sbi/openapi/libogssbi-openapi.so.2.6.4 subprojects/rt-libflute/libflute.a -pthread /usr/lib/x86_64-linux-gnu/libtalloc.so /usr/lib/x86_64-linux-gnu/libyaml.so /usr/lib/x86_64-linux-gnu/libgnutls.so /usr/lib/x86_64-linux-gnu/libssl.so /usr/lib/x86_64-linux-gnu/libcrypto.so /usr/lib/x86_64-linux-gnu/libnghttp2.so /usr/lib/x86_64-linux-gnu/libmicrohttpd.so /usr/lib/x86_64-linux-gnu/libcurl.so -lpthread /usr/lib/x86_64-linux-gnu/libspdlog.so.1.12.0 /usr/lib/x86_64-linux-gnu/libtinyxml2.so /usr/lib/x86_64-linux-gnu/libnl-3.so /usr/lib/x86_64-linux-gnu/libz.so /usr/lib/x86_64-linux-gnu/libfmt.so.9.1.0 -lz -lcrypto /usr/lib/x86_64-linux-gnu/libuuid.so /usr/local/lib/x86_64-linux-gnu/libmpd++.so -lz -lcrypto -Wl,--end-group
/usr/bin/ld: src/mbstf/libmbstf.a.p/DASHManifestHandler.cc.o: in function `com::fiveg_mag::ref_tools::mbstf::DASHManifestHandler::update(std::pair<std::vector<unsigned char, std::allocator<unsigned char> >, com::fiveg_mag::ref_tools::mbstf::ObjectStore::Metadata> const&)':
/home/dsi/5gmag/rt-mbs-transport-function/build/../src/mbstf/DASHManifestHandler.cc:203:(.text+0x16f2): undefined reference to `com::bbc::libmpdpp::MPD::operator==(com::bbc::libmpdpp::MPD const&) const'
collect2: error: ld returned 1 exit status
```

--- a/meson.build
+++ b/meson.build
@@ -9,14 +9,14 @@
 
 # Meson module fs and its functions like fs.hash_file require atleast meson 0.59.0 
 
-project('rt-mbs-transport-function', 'c', 'cpp', 
-    version : '1.2.1',
-    license : '5G-MAG Public',
-    meson_version : '>= 1.4.0',
-    default_options : [
-        'c_std=gnu99',
-        'cpp_std=gnu++20',
-    ],
+project('rt-mbs-transport-function', 'c', 'cpp',
+        version : '1.2.1',
+        license : '5G-MAG Public',
+        meson_version : '>= 1.4.0',
+        default_options : [
+            'c_std=gnu99',
+            'cpp_std=gnu++20',
+        ],
 )
 
 cplusplus = meson.get_compiler('cpp')
@@ -28,17 +28,23 @@ int main(int argc, char *argv[])
   std::chrono::parse("%Y", tmptime);
   return 0;
 }
-''', name: 'STL contains std::chrono::parse')
-  error('std::chrono::parse function not found in STL, please use gcc 14.1 or later as the compiler')
+''', name : 'STL contains std::chrono::parse')
+    error('std::chrono::parse function not found in STL, please use gcc 14.1 or later as the compiler')
 endif
 
-cmake=import('cmake')
+cmake = import('cmake')
 
-open5gs_project=subproject('open5gs', required: true)
+open5gs_project = subproject('open5gs', required : true)
 
-libflute_project=cmake.subproject('rt-libflute')
+# Configure CMake options for the subproject
+libflute_cmake_opts = cmake.subproject_options()
+libflute_cmake_opts.add_cmake_defines({ 'BUILD_TESTING' : 'OFF'})
+libflute_project = cmake.subproject(
+    'rt-libflute',
+    options : libflute_cmake_opts,
+)
 #message('rt-libflute CMake targets:\n - ' + '\n - '.join(libflute_project.target_list()))
-rt_libflute_dep=libflute_project.dependency('flute')
+rt_libflute_dep = libflute_project.dependency('flute')
 
 #subdir('lib')
 subdir('src')

